bika/lims/browser/worksheet/views/analyses.py

```diff
@@ -71,216 +71,361 @@ def __init__(self, context, request):
                         'Attachments']
              },
         ]
+        self.uids_positions = None
+        self.items_rowspans = dict()
+
+    def isItemAllowed(self, obj):
+        """
+        Returns true if the current analysis to be rendered has a slot assigned
+        for the current layout.
+        :param obj: analysis to be rendered as a row in the list
+        :type obj: ATContentType/DexterityContentType
+        :return: True if the obj has an slot assigned. Otherwise, False.
+        :rtype: bool
+        """
+        item_obj = api.get_object(obj)
+        uid = item_obj.UID()
+        if not self.get_item_slot(uid):
+            logger.warning("Slot not assigned to item %s" % uid)
+            return False
+        return BaseView.isItemAllowed(self, obj)
+
+    def folderitem(self, obj, item, index):
+        """
+        Applies new properties to the item (analysis) that is currently being
+        rendered as a row in the list.
+        :param obj: analysis to be rendered as a row in the list
+        :param item: dict representation of the analysis, suitable for the list
+        :param index: current position of the item within the list
+        :type obj: ATContentType/DexterityContentType
+        :type item: dict
+        :type index: int
+        :return: the dict representation of the item
+        :rtype: dict
+        """
+        item = BaseView.folderitem(self, obj, item, index)
+        item_obj = api.get_object(obj)
+        uid = item['uid']
+
+        # Slot is the row position where all analyses sharing the same parent
+        # (eg. AnalysisRequest, SampleReference), will be displayed as a group
+        slot = self.get_item_slot(uid)
+        item['Pos'] = slot
+
+        # The position string contains both the slot + the position of the
+        # analysis within the slot: "position_sortkey" will be used to sort all
+        # the analyses to be displayed in the list
+        str_position = self.get_uids_strpositions()[uid]
+        item['pos_sortkey'] = str_position
+
+        item['colspan'] = {'Pos':1}
+        item['Service'] = item_obj.Title()
+        item['Category'] = item_obj.getCategoryTitle()
+        item['DueDate'] = self.ulocalized_time(item_obj, long_format=0)
+        item['class']['Service'] = 'service_title'
+
+        # To prevent extra loops, we compute here the number of analyses to be
+        # rendered within each slot. This information will be useful later for
+        # applying rowspan to the first cell of each slot, that contains info
+        # about the parent of all the analyses contained in that slot (e.g
+        # Analysis Request ID, Sample Type, etc.)
+        rowspans = self.items_rowspans.get(slot, 0) + 1
+        remarks = item.get('Remarks', None)
+        remarks_edit = 'Remarks' in item.get('allow_edit', [])
+        remarks_enabled = self.context.bika_setup.getEnableAnalysisRemarks()
```

Comment:
put `bika_setup` also in `__call__` like this: `self.bika_setup = api.get_bika_setup()`. So we avoid this object acquisition lookup each time